### DISCUSSION TITLE
Accessing texture IDs to pass to ImGui

### DISCUSSION THREAD

**Discussion by josefgraus** [markaren/threepp#145] [Q&A] [answered]
I've run into a problem I can't immediately and cleanly solve.

`ImGui::ImageButton()` will display an icon that has an image drawn on it as a GUI element. However, it explicitly requires that the image be supplied as a texture ID from whichever graphics API that is being used to render into the context that ImGui is drawing over. I've dug into the interface for various parts of threepp (Texture, Image, Mesh, Material, Scene, etc..) but haven't been able to find a way to cleanly query the texture ID allocated to an image loaded via `TextureLoader`. As far as I can tell, it's not until `Renderer::render()` is called that these textures are generated and associated with the loaded image data.

Is there a preferred or design-friendly envisioned way to query the underlying texture ID associated with a loaded `Texture`?

**Comment by markaren** [comment, marked as answer]
I guess this would have to go through the GLRenderer as the overall API is rendering agnostic, but have not given it much thought.

**Reply by josefgraus** [reply]
No worries. I just thought I'd check. It's something of an annoying case since having that nice agnostic abstraction is definitely preferred. For now I'll just use `glad` to manually generate and bind texture IDs for icons since I'm currently only targeting opengl for this demo app.

Thanks for the quick response!

**Comment by markaren** [comment]
Accessing texture IDs is now possible. See #186
